R/package-auth.R

```diff
@@ -78,23 +78,23 @@ db_host <- function(id = NULL, prefix = NULL, profile = getOption("db_profile",
 #' @import cli
 #' @export
 db_token <- function(profile = getOption("db_profile")) {
+  use_databrickscfg <- getOption("use_databrickscfg", FALSE)
+
+  # Check if running within RStudio on Posit Workbench so managed OAuth credentials get used
+  if ((Sys.getenv("DATABRICKS_CONFIG_PROFILE") == "workbench") && (Sys.getenv("DATABRICKS_TOKEN") != "")) {
+    use_databrickscfg <- TRUE
+    profile <- "workbench"
```

Comment:
What I'd suggest here is introducing a helper function to look up the config profile:

```r
default_config_profile <- function() {
  profile <- Sys.getenv("DATABRICKS_CONFIG_PROFILE")
  if (nchar(profile) != 0) {
    profile
  } else {
    getOption("db_profile")
  }
}
```

and then modifying `db_token()` so that it takes `profile = default_config_profile()`. And finally, making this use

```r
use_databricks_cfg <- getOption("use_databrickscfg", FALSE)
if (!is.null(profile)) {
  use_databricks_cfg <- TRUE
}
```

in this case: (1) setting a profile enables reading the config, which kind of makes sense anyway; (2) the environment variable can be documented as taking precedence over the option, which also makes sense to me.

Reply:
Good ideas. I was initially trying to preserve the original order.

The one challenging thing is the `brickster` package actually does allow people to set a `profile` and have that be a work added on the end of `DATABRICKS_TOKEN` and `DATABRICKS_HOST`, so automatically assuming it is the config file would break this. For this reason, I think it's still good to check if the profile is `workbench` and make a special case to set.